wrapper/src/main/java/software/amazon/jdbc/ConnectionPluginManager.java

```diff
@@ -505,6 +507,36 @@ public HostSpec getHostSpecByStrategy(HostRole role, String strategy)
     }
   }
 
+  public HostSpec getHostSpecByStrategy(List<HostSpec> hosts, HostRole role, String strategy)
+      throws SQLException, UnsupportedOperationException {
+    try {
+      for (ConnectionPlugin plugin : this.plugins) {
+        Set<String> pluginSubscribedMethods = plugin.getSubscribedMethods();
+        boolean isSubscribed =
+            pluginSubscribedMethods.contains(ALL_METHODS)
+                || pluginSubscribedMethods.contains(GET_HOST_SPEC_BY_STRATEGY_METHOD);
+
+        if (isSubscribed) {
+          try {
+            final HostSpec host = plugin.getHostSpecByStrategy(hosts, role, strategy);
+            if (host != null) {
+              return host;
+            }
+          } catch (UnsupportedOperationException e) {
+            // This plugin does not support the provided strategy, ignore the exception and move on
+          }
+        }
+      }
+
+      throw new UnsupportedOperationException(
+          "The driver does not support the requested host selection strategy: " + strategy);
+    } catch (RuntimeException e) {
+      throw e;
```

Comment:
Reminder to remove this if this is for debugging. Unless the goal is to wrap the runtime exception into another exception?

Reply:
Thanks for catching this. I think this is was a bad copy n paste job from the original method. I'll remove it from both.